src/libpriv/rpmostree-core.c

```diff
@@ -2015,14 +2157,11 @@ rpmostree_context_assemble_commit (RpmOstreeContext      *self,
                                                             pkg),
                                        cancellable, error))
         goto out;
-    }
 
-  if (filesystem_package)
-    i = 0;
-  else
-    i = 1;
```

Comment:
Is it really OK to ditch this logic?  Won't we be unpacking the filesystem package twice?